WebIDL: Use `Uint8ClampedArray` instead of raw `JSObject` in bindings

Taym95 opened the pull request:
<!-- Please describe your changes on the following line: -->
part #31064

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [ ] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

<!-- Either: -->
- [x] These changes do not require tests because there is not behavior change

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mrobinson committed:
deps: Stop vendoring WebRender (#31212)

The new strategy for dependencies with upstream in Gecko is to manage
them in separate repositories, which will more easily allow rebasing our
changes on top of newer Gecko work.

Taym95 committed:
webidlg: Handle `Float64Array` as a `TypedArray` rather than a raw `JSObject` (#31189)

* WebIDL use FLoat64Array

Signed-off-by: Bentaimia Haddadi <haddadi.taym@gmail.com>

* Use to_vec to convert array to vec

* avoid allocating a new vec

---------

Signed-off-by: Bentaimia Haddadi <haddadi.taym@gmail.com>

wusyong committed:
Document media configs from prefs (#31223)

Loirooriol committed:
Lint layout_2013 with clippy (#31221)

* Lint layout_2013 with clippy

CARGO_BUILD_RUSTC=rustc cargo clippy --fix -p layout_2013 --broken-code

* ./mach fmt

* Cosmetic adjustments

mrobinson committed:
mach: Error out sooner with Python < 3.10 (#31225)

We upgraded our Python dependency to 3.10, but neglected to update mach.
This change makes sure we error out sooner, rather than later.

mrobinson committed:
dependencies: Upgrade surfman to 0.9 (#31224)

This upgrades surfman and webxr. The main benefit to this upgrade is
that surfman (and thus libservo) no longer depends on winit. servoshell
still does, but this should make upgrades a lot easier.

Loirooriol committed:
Allow using cached client_rect() for paint-only reflow (#31219)

PR #31210 avoided the cache for all kinds of reflow, but it's actually
fine to use it for ReflowTriggerCondition::PaintPostponed.

Taym95 committed:
WebIDL: Use Uint8ClampedArray instead of raw JSObject in bindings

dependabot[bot] committed:
build(deps): bump profiling from 1.0.13 to 1.0.14 (#31227)

Bumps [profiling](https://github.com/aclysma/profiling) from 1.0.13 to 1.0.14.
- [Changelog](https://github.com/aclysma/profiling/blob/master/CHANGELOG.md)
- [Commits](https://github.com/aclysma/profiling/compare/v1.0.13...v1.0.14)

---
updated-dependencies:
- dependency-name: profiling
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mukilan committed:
Fix bugs in `mach test-tidy` (#31232)

* mach: test-tidy should return 1 for failure

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

* fix lint issues

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

* mach: tidy should allow spec links with trailing text

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

---------

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

dependabot[bot] committed:
build(deps): bump itertools from 0.12.0 to 0.12.1 (#31229)

Bumps [itertools](https://github.com/rust-itertools/itertools) from 0.12.0 to 0.12.1.
- [Changelog](https://github.com/rust-itertools/itertools/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rust-itertools/itertools/compare/v0.12.0...v0.12.1)

---
updated-dependencies:
- dependency-name: itertools
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump darling from 0.20.3 to 0.20.4 (#31228)

Bumps [darling](https://github.com/TedDriggs/darling) from 0.20.3 to 0.20.4.
- [Release notes](https://github.com/TedDriggs/darling/releases)
- [Changelog](https://github.com/TedDriggs/darling/blob/master/CHANGELOG.md)
- [Commits](https://github.com/TedDriggs/darling/compare/v0.20.3...v0.20.4)

---
updated-dependencies:
- dependency-name: darling
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump darling from 0.20.4 to 0.20.5 (#31237)

Bumps [darling](https://github.com/TedDriggs/darling) from 0.20.4 to 0.20.5.
- [Release notes](https://github.com/TedDriggs/darling/releases)
- [Changelog](https://github.com/TedDriggs/darling/blob/master/CHANGELOG.md)
- [Commits](https://github.com/TedDriggs/darling/commits)

---
updated-dependencies:
- dependency-name: darling
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump synstructure from 0.13.0 to 0.13.1 (#31239)

Bumps [synstructure](https://github.com/mystor/synstructure) from 0.13.0 to 0.13.1.
- [Commits](https://github.com/mystor/synstructure/commits)

---
updated-dependencies:
- dependency-name: synstructure
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Taym95 commented on components/script/dom/imagedata.rs:
```diff
@@ -131,7 +134,7 @@ impl ImageData {
         height: u32,
     ) -> Fallible<DomRoot<Self>> {
-        unsafe { Self::new_without_jsobject(global, proto, width, height) }
+        unsafe { Self::new_without_data(global, proto, width, height) }
```
@gterzian can you review this when you got time, still in progress

dependabot[bot] committed:
build(deps): bump webxr from `f1cc785` to `614420b` (#31238)

Bumps [webxr](https://github.com/servo/webxr) from `f1cc785` to `614420b`.
- [Commits](https://github.com/servo/webxr/compare/f1cc78521616883ca1592129848b525dcf8b4136...614420b9830615376563fc6e1d98c52119f97123)

---
updated-dependencies:
- dependency-name: webxr
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gterzian commented on components/script/dom/imagedata.rs:
```diff
@@ -111,15 +111,18 @@ impl ImageData {
-        Uint8ClampedArray::create(*cx, CreateWith::Length(len as usize), array.handle_mut())
-            .unwrap();
-        (*imagedata).data.set(array.get());
+        let typed_array =
```
I think this belongs in a new `new_initialized` method of the heaped array, and you should create it above before `imagedata`.

gterzian replied:
And you can make `new_initialized` take an optional `JSObject`, so that it can be used with `new_with_js_object`(see below).

gterzian commented on components/script/dom/imagedata.rs:
```diff
@@ -144,23 +147,23 @@ impl ImageData {
     ) -> Fallible<DomRoot<Self>> {
-        Self::new_with_jsobject(global, proto, width, opt_height, jsobject)
+        // Self::new_with_data(global, proto, width, opt_height, jsobject)
+        Self::new_without_data(global, proto, width, opt_height.unwrap())
```
Here you should keep `new_with_jsobject`, and create a new `new_initialized` method on `HeapTypedArray` that takes an optional jsobject, in this case some.

gterzian reviewed (changes requested):
Thanks! This one is another rather complicated one, so I suggest a few changes.

dependabot[bot] committed:
build(deps): bump app_units from 0.7.3 to 0.7.4 (#31244)

Bumps [app_units](https://github.com/servo/app_units) from 0.7.3 to 0.7.4.
- [Release notes](https://github.com/servo/app_units/releases)
- [Commits](https://github.com/servo/app_units/compare/v0.7.3...v0.7.4)

---
updated-dependencies:
- dependency-name: app_units
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump winnow from 0.5.35 to 0.5.36 (#31243)

Bumps [winnow](https://github.com/winnow-rs/winnow) from 0.5.35 to 0.5.36.
- [Changelog](https://github.com/winnow-rs/winnow/blob/main/CHANGELOG.md)
- [Commits](https://github.com/winnow-rs/winnow/compare/v0.5.35...v0.5.36)

---
updated-dependencies:
- dependency-name: winnow
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump indexmap from 2.2.1 to 2.2.2 (#31245)

Bumps [indexmap](https://github.com/indexmap-rs/indexmap) from 2.2.1 to 2.2.2.
- [Changelog](https://github.com/indexmap-rs/indexmap/blob/master/RELEASES.md)
- [Commits](https://github.com/indexmap-rs/indexmap/compare/2.2.1...2.2.2)

---
updated-dependencies:
- dependency-name: indexmap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

syvb committed:
Make Android build optional on Nix (#31231)

* Make Android build optional on Nix

I want to build Servo without also installing the entire Android SDK.
This makes it so Android support is only built if `buildAndroid` is
true.

Signed-off-by: syvb <me@iter.ca>

* Add Android support to nix-shell if SERVO_ANDROID_BUILD set

---------

Signed-off-by: syvb <me@iter.ca>

dependabot[bot] committed:
build(deps): bump libc from 0.2.152 to 0.2.153 (#31248)

Bumps [libc](https://github.com/rust-lang/libc) from 0.2.152 to 0.2.153.
- [Release notes](https://github.com/rust-lang/libc/releases)
- [Commits](https://github.com/rust-lang/libc/compare/0.2.152...0.2.153)

---
updated-dependencies:
- dependency-name: libc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump rustix from 0.38.30 to 0.38.31 (#31249)

Bumps [rustix](https://github.com/bytecodealliance/rustix) from 0.38.30 to 0.38.31.
- [Release notes](https://github.com/bytecodealliance/rustix/releases)
- [Commits](https://github.com/bytecodealliance/rustix/compare/v0.38.30...v0.38.31)

---
updated-dependencies:
- dependency-name: rustix
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump webpki-roots from 0.25.3 to 0.25.4 (#31250)

Bumps [webpki-roots](https://github.com/rustls/webpki-roots) from 0.25.3 to 0.25.4.
- [Release notes](https://github.com/rustls/webpki-roots/releases)
- [Commits](https://github.com/rustls/webpki-roots/compare/v/0.25.3...v/0.25.4)

---
updated-dependencies:
- dependency-name: webpki-roots
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump app_units from 0.7.4 to 0.7.5 (#31252)

Bumps [app_units](https://github.com/servo/app_units) from 0.7.4 to 0.7.5.
- [Release notes](https://github.com/servo/app_units/releases)
- [Commits](https://github.com/servo/app_units/compare/v0.7.4...v0.7.5)

---
updated-dependencies:
- dependency-name: app_units
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

syvb committed:
layout: return None bounding box when no nodes found (#31253)

Signed-off-by: syvb <me@iter.ca>

dependabot[bot] committed:
build(deps): bump tokio from 1.35.1 to 1.36.0 (#31251)

Bumps [tokio](https://github.com/tokio-rs/tokio) from 1.35.1 to 1.36.0.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-1.35.1...tokio-1.36.0)

---
updated-dependencies:
- dependency-name: tokio
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrobinson committed:
libservo: Handle GL video decoding setup internally (#31209)

Instead of making the client set up GL video decoding, have this done
inside libservo. The details necessary for decoding are fetched via
Surfman now. This also removes the setup for the context on Android --
which has no GStreamer support now anyway. In the future, this could be
enabled, but should likely be done using Surfman, instead of passing on
all these details manually.

syvb committed:
Convert etc/profilicate.py to Python 3 (#31259)

servo-wpt-sync committed:
Sync WPT with upstream (04-02-2024) (#31256)

* Update web-platform-tests to revision b'faa6d42ce21d126155dee771786ad7c74efbcba5'

* Update test expectation

---------

Co-authored-by: cybai <cyb.ai.815@gmail.com>

syvb committed:
canvas2d: Implement `.reset()` (#31258)

* Implement Canvas2D reset

* Update WPT tests

* Apply suggestions from code review

---------

Co-authored-by: Martin Robinson <mrobinson@igalia.com>

jdm committed:
Initial overview of webxr initialization (#31255)

syvb committed:
Make console methods take `any` instead of `string` (#31241)

* Make console methods use `any` for the message

Match the Console spec by allowing any value to be passed to console
methods, instead of just values that can be converted to a string.

Signed-off-by: syvb <me@iter.ca>

* Add test for console logging a Symbol

Signed-off-by: syvb <me@iter.ca>

* Implement object stringification for logs

Signed-off-by: syvb <me@iter.ca>

* Address review comments

Signed-off-by: syvb <me@iter.ca>

* Make time/timeEnd accept DOMString to match spec

* Update WPT results for layout 2013

---------

Signed-off-by: syvb <me@iter.ca>

dependabot[bot] committed:
build(deps): bump miniz_oxide from 0.7.1 to 0.7.2 (#31260)

Bumps [miniz_oxide](https://github.com/Frommi/miniz_oxide) from 0.7.1 to 0.7.2.
- [Changelog](https://github.com/Frommi/miniz_oxide/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Frommi/miniz_oxide/commits)

---
updated-dependencies:
- dependency-name: miniz_oxide
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump winnow from 0.5.36 to 0.5.37 (#31263)

Bumps [winnow](https://github.com/winnow-rs/winnow) from 0.5.36 to 0.5.37.
- [Changelog](https://github.com/winnow-rs/winnow/blob/main/CHANGELOG.md)
- [Commits](https://github.com/winnow-rs/winnow/compare/v0.5.36...v0.5.37)

---
updated-dependencies:
- dependency-name: winnow
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump iana-time-zone from 0.1.59 to 0.1.60 (#31266)

Bumps [iana-time-zone](https://github.com/strawlab/iana-time-zone) from 0.1.59 to 0.1.60.
- [Changelog](https://github.com/strawlab/iana-time-zone/blob/main/CHANGELOG.md)
- [Commits](https://github.com/strawlab/iana-time-zone/compare/v0.1.59...v0.1.60)

---
updated-dependencies:
- dependency-name: iana-time-zone
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrobinson committed:
layout: Collect both start and end baselines for fragments (#31230)

This change starts collecting the starting baseline set for fragments,
which is necessary for some layout modes (flex and tables, namely) as
well as being important for the implementation of `align-items`. In
addition, it converts baseline measurement to use `Au` everywhere.

Co-authored-by: Oriol Brufau <obrufau@igalia.com>

dependabot[bot] committed:
build(deps): bump exr from 1.71.0 to 1.72.0 (#31264)

Bumps [exr](https://github.com/johannesvollmer/exrs) from 1.71.0 to 1.72.0.
- [Release notes](https://github.com/johannesvollmer/exrs/releases)
- [Commits](https://github.com/johannesvollmer/exrs/commits)

---
updated-dependencies:
- dependency-name: exr
  dependency-type: indirect
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

syvb committed:
Fix crash on large console log (#31267)

mrobinson committed:
layout: Wait to count justification opportunities until justification (#31236)

Instead of tracking justification opportunities during line layout, wait
until the line is about to be laid out and justification is about
happen. This makes the logic for tracking justification opportunities
simpler. In particular, we no longer have to carefully adjust them when
trimming whitespace. Additionally, this avoids a bit of work unless
justification is turned on.

This also includes a small cleanup of the justification code.

dependabot[bot] committed:
build(deps): bump hermit-abi from 0.3.4 to 0.3.5 (#31273)

Bumps [hermit-abi](https://github.com/hermitcore/hermit-rs) from 0.3.4 to 0.3.5.
- [Release notes](https://github.com/hermitcore/hermit-rs/releases)
- [Commits](https://github.com/hermitcore/hermit-rs/compare/hermit-abi-0.3.4...hermit-abi-0.3.5)

---
updated-dependencies:
- dependency-name: hermit-abi
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump half from 2.2.1 to 2.3.1 (#31275)

Bumps [half](https://github.com/starkat99/half-rs) from 2.2.1 to 2.3.1.
- [Release notes](https://github.com/starkat99/half-rs/releases)
- [Changelog](https://github.com/starkat99/half-rs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/starkat99/half-rs/compare/v2.2.1...v2.3.1)

---
updated-dependencies:
- dependency-name: half
  dependency-type: indirect
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump tempfile from 3.9.0 to 3.10.0 (#31274)

Bumps [tempfile](https://github.com/Stebalien/tempfile) from 3.9.0 to 3.10.0.
- [Changelog](https://github.com/Stebalien/tempfile/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Stebalien/tempfile/compare/v3.9.0...v3.10.0)

---
updated-dependencies:
- dependency-name: tempfile
  dependency-type: indirect
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mukilan committed:
android: disable JIT in SM to workaround #31134 (#31270)

The crash when loading servo.org happens in the JIT code
emitted by SM's CacheIRCompiler to invoke the VM function
`ProxyGetPropertyByValue`.

To disable this code path, it is not sufficient to disable
just the baseline JIT (which exposed in servo under the
pref `js.baseline.enabled`) but also the baseline
interpreter which is controlled by a different flag in SM.

This PR disables renames the `js.baseline.enabled` pref in
Servo to `js.baseline_jit.enabled` and introduces a new
pref `js.baseline_interpreter.enabled` that controls the
baseline interpreter.

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

atbrakhi committed:
Fix ./mach bootstrap failure in debian (#31276)

Do not install `libgstreamer-plugins-good1.0-dev` on debian,
install only on ubuntu.

atbrakhi committed:
Remove duplicate pkg in APT_PKGS (#31280)

We have `libgstreamer-plugins-base1.0-dev` twice in APT_PKGS,
looks like a duplicate, this PR removes the duplicated pkg.

mrobinson committed:
bootstrap: More resiliently install Deiban-like platform dependencies (#31281)

1. First check to see if a package is available before trying to install
   it. This means that we always do our best to install everything, but
   don't fail if we cannot.
2. Install crown and taplo first. This means that if the
   platform-specific bits fail, we still install Servo-specific
   dependencies.

dependabot[bot] committed:
build(deps): bump libgit2-sys from 0.16.1+1.7.1 to 0.16.2+1.7.2 (#31283)

Bumps [libgit2-sys](https://github.com/rust-lang/git2-rs) from 0.16.1+1.7.1 to 0.16.2+1.7.2.
- [Changelog](https://github.com/rust-lang/git2-rs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rust-lang/git2-rs/commits)

---
updated-dependencies:
- dependency-name: libgit2-sys
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump js-sys from 0.3.67 to 0.3.68 (#31284)

Bumps [js-sys](https://github.com/rustwasm/wasm-bindgen) from 0.3.67 to 0.3.68.
- [Release notes](https://github.com/rustwasm/wasm-bindgen/releases)
- [Changelog](https://github.com/rustwasm/wasm-bindgen/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rustwasm/wasm-bindgen/commits)

---
updated-dependencies:
- dependency-name: js-sys
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump bytemuck from 1.14.1 to 1.14.2 (#31285)

Bumps [bytemuck](https://github.com/Lokathor/bytemuck) from 1.14.1 to 1.14.2.
- [Changelog](https://github.com/Lokathor/bytemuck/blob/main/changelog.md)
- [Commits](https://github.com/Lokathor/bytemuck/compare/v1.14.1...v1.14.2)

---
updated-dependencies:
- dependency-name: bytemuck
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump git2 from 0.18.1 to 0.18.2 (#31282)

Bumps [git2](https://github.com/rust-lang/git2-rs) from 0.18.1 to 0.18.2.
- [Changelog](https://github.com/rust-lang/git2-rs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rust-lang/git2-rs/compare/git2-0.18.1...git2-0.18.2)

---
updated-dependencies:
- dependency-name: git2
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump winnow from 0.5.37 to 0.5.39 (#31286)

Bumps [winnow](https://github.com/winnow-rs/winnow) from 0.5.37 to 0.5.39.
- [Changelog](https://github.com/winnow-rs/winnow/blob/main/CHANGELOG.md)
- [Commits](https://github.com/winnow-rs/winnow/compare/v0.5.37...v0.5.39)

---
updated-dependencies:
- dependency-name: winnow
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump unicode-segmentation from 1.10.1 to 1.11.0 (#31288)

Bumps [unicode-segmentation](https://github.com/unicode-rs/unicode-segmentation) from 1.10.1 to 1.11.0.
- [Commits](https://github.com/unicode-rs/unicode-segmentation/compare/v1.10.1...v1.11.0)

---
updated-dependencies:
- dependency-name: unicode-segmentation
  dependency-type: indirect
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

syvb committed:
script: implement navigator.hardwareConcurrency (#31268)

Signed-off-by: syvb <me@iter.ca>

mukilan committed:
mach: fix shell.nix to actually include android pkgs (#31279)

Fix issue with precedence due to which the android
environment variables were being added to the output
of mkDerivation instead of the input attribute set.

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mrobinson committed:
Remove unused `dependencyci.yml` file (#31291)

I'm fairly certain this file is unused and the bugs linked are now closed.

mrobinson committed:
mach: Make `./mach try` a little friendlier (#31290)

1. Move `./mach try` to `testing_commands.py which is a bit more
   consistent.
2. Make `./mach try` print out the remote name always and properly
   form the URL for ssh repositories.
3. Print out the try configuration matrix to make it more obvious
   what is being triggered remotely.
4. Better error handling. Print and error and exit if the remote isn't
   on GitHub and also clean up properly if something fails after making
   the temporary commit.

dependabot[bot] committed:
build(deps): bump web-sys from 0.3.67 to 0.3.68 (#31294)

Bumps [web-sys](https://github.com/rustwasm/wasm-bindgen) from 0.3.67 to 0.3.68.
- [Release notes](https://github.com/rustwasm/wasm-bindgen/releases)
- [Changelog](https://github.com/rustwasm/wasm-bindgen/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rustwasm/wasm-bindgen/commits)

---
updated-dependencies:
- dependency-name: web-sys
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump num-integer from 0.1.45 to 0.1.46 (#31295)

Bumps [num-integer](https://github.com/rust-num/num-integer) from 0.1.45 to 0.1.46.
- [Changelog](https://github.com/rust-num/num-integer/blob/master/RELEASES.md)
- [Commits](https://github.com/rust-num/num-integer/compare/num-integer-0.1.45...num-integer-0.1.46)

---
updated-dependencies:
- dependency-name: num-integer
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump jobserver from 0.1.27 to 0.1.28 (#31296)

Bumps [jobserver](https://github.com/alexcrichton/jobserver-rs) from 0.1.27 to 0.1.28.
- [Commits](https://github.com/alexcrichton/jobserver-rs/commits)

---
updated-dependencies:
- dependency-name: jobserver
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump num-traits from 0.2.17 to 0.2.18 (#31293)

Bumps [num-traits](https://github.com/rust-num/num-traits) from 0.2.17 to 0.2.18.
- [Changelog](https://github.com/rust-num/num-traits/blob/master/RELEASES.md)
- [Commits](https://github.com/rust-num/num-traits/compare/num-traits-0.2.17...num-traits-0.2.18)

---
updated-dependencies:
- dependency-name: num-traits
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrobinson committed:
layout: Use `BoxFragment` border widths for display list generation (#31292)

Instead of using the border widths from the style, use the ones recorded
by the `BoxFragment`. This is necessary because inline layout can
override these border widths during fragmentation. For instance, when a
box is split across two lines only one fragment should have an inline
start border.

rajveermalviya committed:
BHM: Support aarch64 for Mac sampler (#31298)

Replaced unmaintained `mach` crate with `mach2`, and added support for getting arm registers.

Loirooriol committed:
Fix CSSStyleDeclaration's item() to provide properties instead of values (#31299)

For example,

    style.cssText = "margin: 1px 2px 3px 4px";
    [...style];

will now be

    ["margin-top", "margin-right", "margin-bottom", "margin-left"]

instead of

    ["1px", "2px", "3px", "4px"]

dependabot[bot] committed:
build(deps): bump bytemuck from 1.14.2 to 1.14.3 (#31301)

Bumps [bytemuck](https://github.com/Lokathor/bytemuck) from 1.14.2 to 1.14.3.
- [Changelog](https://github.com/Lokathor/bytemuck/blob/main/changelog.md)
- [Commits](https://github.com/Lokathor/bytemuck/compare/v1.14.2...v1.14.3)

---
updated-dependencies:
- dependency-name: bytemuck
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Loirooriol committed:
Don't escape property name in CSSStyleDeclaration's item() (#31305)

Just return the raw name. This is only relevant for custom properties.

mrobinson committed:
Update mozangle and mozjs in order to use bindgen 0.69.4 (#31306)

This should fix issues that we have with Float16 support on newer LLVM.
This also updates style, in order to remove the duplicate version of
bindgen. We will soon be able to manage updates to style more
consistently.

dependabot[bot] committed:
build(deps): bump cfg-expr from 0.15.6 to 0.15.7 (#31303)

Bumps [cfg-expr](https://github.com/EmbarkStudios/cfg-expr) from 0.15.6 to 0.15.7.
- [Release notes](https://github.com/EmbarkStudios/cfg-expr/releases)
- [Changelog](https://github.com/EmbarkStudios/cfg-expr/blob/main/CHANGELOG.md)
- [Commits](https://github.com/EmbarkStudios/cfg-expr/compare/0.15.6...0.15.7)

---
updated-dependencies:
- dependency-name: cfg-expr
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump is-terminal from 0.4.10 to 0.4.11 (#31302)

Bumps [is-terminal](https://github.com/sunfishcode/is-terminal) from 0.4.10 to 0.4.11.
- [Commits](https://github.com/sunfishcode/is-terminal/compare/v0.4.10...v0.4.11)

---
updated-dependencies:
- dependency-name: is-terminal
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrobinson committed:
layout: Start work on table row height and vertical-align (#31246)

This implements a very naive row height allocation approach. It has just
enough to implement `vertical-align` in table cells. Rowspanned cells
get enough space for their content, with the extra space necessary being
allocated to the last row. There's still a lot missing here, including
proper distribution of row height to rowspanned cells.

Co-authored-by: Oriol Brufau <obrufau@igalia.com>

sebsebmc committed:
<hr> elements are expected to have a default overflow:hidden (#31297)

* <hr> elements are expected to have a default overflow:hidden
See https://github.com/whatwg/html/pull/2724

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

* Use full defined hr style

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

* update legacy test expectation

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

---------

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

sebsebmc committed:
layout: Respond to shorthand property requests with real values (#31277)

* Respond to shorthand property requests with real values

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

* Cleanup formatting and old comments

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

* Update WPT expectations

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

* Refactor out helper fn

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

* Cleanup

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

---------

Signed-off-by: Sebastian C <sebsebmc@gmail.com>

servo-wpt-sync committed:
Sync WPT with upstream (11-02-2024) (#31309)

* Update web-platform-tests to revision b'33d11f1db34802fda00e64ddeb0b7ef040cf65be'

* Update expectations

---------

Co-authored-by: sagudev <16504129+sagudev@users.noreply.github.com>

syvb committed:
Explictly specify num_cpus version (#31310)

This fixes a warning from Cargo about relying on deprecated behavior:
```
warning: /home/loops/src/servo/servo/components/script/Cargo.toml: dependency (num_cpus) specified without providing a local path, Git repository, version, or workspace dependency to use. This will be considered an error in future versions
warning: /home/loops/src/servo/servo/components/config/Cargo.toml: dependency (num_cpus) specified without providing a local path, Git repository, version, or workspace dependency to use. This will be considered an error in future versions
```

mrobinson committed:
layout: Implement support for font fallback (#31254)

This adds an initial implementation of font fallback, on part with the
one used in legacy layout. There are still issues. For instance, font
matching is done per unicode character rather than based on graphemes or
the shape first approach of Chrome. The idea is that these changes can
be made later.

Taym95 committed:
WebIDL: Use Uint8ClampedArray instead of raw JSObject in bindings

Signed-off-by: Bentaimia Haddadi <haddadi.taym@gmail.com>

servo-wpt-sync commented:
🛠 These changes could not be applied onto the latest upstream WPT. Servo's copy of the Web Platform Tests may be out of sync.
